backends/gdx-backends-gwt/src/com/badlogic/gwtref/client/ReflectionCache.java

```diff
@@ -44,35 +44,47 @@ private static String convert (String className) {
 		if (className.startsWith("[")) {
 			int dimensions = 0;
 			char c = className.charAt(0);
-			String suffix = "";
+			StringBuilder stringBuilder = new StringBuilder();
 			while (c == '[') {
 				dimensions++;
-				suffix += "[]";
 				c = className.charAt(dimensions);
 			}
 			char t = className.charAt(dimensions);
 			switch (t) {
 			case 'Z':
-				return "boolean" + suffix;
+				stringBuilder.append("boolean");
+				break;
 			case 'B':
-				return "byte" + suffix;
+				stringBuilder.append("byte");
+				break;
 			case 'C':
-				return "char" + suffix;
+				stringBuilder.append("char");
+				break;
 			case 'L':
-				return className.substring(dimensions + 1, className.length() - 1).replace('$', '.') + suffix;
+				stringBuilder.append(className.substring(dimensions + 1, className.length() - 1).replace('$', '.'));
+				break;
 			case 'D':
-				return "double" + suffix;
+				stringBuilder.append("double");
+				break;
 			case 'F':
-				return "float" + suffix;
+				stringBuilder.append("float");
+				break;
 			case 'I':
-				return "int" + suffix;
+				stringBuilder.append("int");
+				break;
 			case 'J':
-				return "long" + suffix;
+				stringBuilder.append("long");
+				break;
 			case 'S':
-				return "short" + suffix;
+				stringBuilder.append("short");
+				break;
 			default:
 				throw new IllegalArgumentException("Couldn't transform '" + className + "' to qualified source name");
 			}
+			for(int i=0;i<dimensions;i++){
+				stringBuilder.append("[]");
+			}
+			return stringBuilder.toString();

```

Comment:
I think this makes it actually more readable, because previously the code appended and prepended to the same `String` variable, now it is just straight up appends. I'd probably rename the `stringBuilder` to something either more relevant (`converted`) or short (`sb`).